### ISSUE TITLE
Please upgrade to socket.io 2.0.1

### ISSUE THREAD

**Issue by jadekler** [karma-runner/karma#2735]
So that you can have a version of `socket.io` which depends on a version of `engine.io` which depends on a version of `ws` that is not horribly out of date. :)

For reference, if you currently have latest version of `karma`, `ws`, and `bufferutil` (which is an optional dependency of `ws`), you'll get a runtime error running karma since the version of bufferutil is not supported by the `socket.io`->`engine.io`->`ws`.

```
31 05 2017 11:19:13.879:ERROR [karma]: TypeError: Cannot read property 'unmask' of undefined
```

**Comment by gdyrrahitis** [comment]
Same error
```
04 06 2017 20:44:56.932:ERROR [karma]: [TypeError: Cannot read property 'unmask' of undefined]
```
Please fix

**Comment by yoieh** [comment]
This is still not done?

I have generated an new angular project with angular cli. The project generated uses Karma and when Im trying to use `ng test` it runs karma with error:

```
01 11 2017 14:03:37.583:ERROR [karma]: TypeError: Cannot read property 'unmask' of undefined
```

I had it working just some weeks ago on an other computer but here it dose not work at all.

from angular projects packages.json:

```
"karma": "~1.7.0",
"karma-chrome-launcher": "~2.1.1",
"karma-cli": "~1.0.1",
"karma-coverage-istanbul-reporter": "^1.2.1",
"karma-jasmine": "~1.1.0",
"karma-jasmine-html-reporter": "^0.2.2",
```

npm-v 
```
5.5.1
```
node -v
```
v8.9.0
```

```
ng -v
```
@angular/cli: 1.4.9
node: 8.9.0
os: win32 x64
@angular/animations: 4.4.6
@angular/common: 4.4.6
@angular/compiler: 4.4.6
@angular/core: 4.4.6
@angular/forms: 4.4.6
@angular/http: 4.4.6
@angular/platform-browser: 4.4.6
@angular/platform-browser-dynamic: 4.4.6
@angular/router: 4.4.6
@angular/cli: 1.4.9
@angular/compiler-cli: 4.4.6
@angular/language-service: 4.4.6
typescript: 2.3.4
```

**Comment by viridia** [comment]
Same here - my tests are now completely broken because the ancient version of ws no longer works.

**Comment by skylarjhdownes** [comment]
https://github.com/karma-runner/karma/pull/2880 should fix this issue.

**Comment by johnjbarton** [comment]
Can anyone explain why this suddenly became a problem?  It sounds like karma has the wrong version of 'ws' for socket.io v1.x?

**Comment by yoieh** [comment]
Yes that's the problem

**Comment by yoieh** [comment]
It's actually cost me a job.

Personly I think the problem is that the TLS version Node doesn't support the version of 'ws' karma uses. A pull request was done but didn't survival testing.

**Comment by johnjbarton** [comment]
Fixed by #2880

**Comment by skylarjhdownes** [comment]
This is fixed in the master branch of Karma, but I don't believe it's been rolled out to the npm install yet.  Is there anything holding that up?  I'd be happy to help if someone could point me in the right direction.
